### PR TITLE
[Finishes #167598393] Add search to Mirage

### PR DESCRIPTION
**Pull request by alexrockets**
#### What does this PR do?

Adds search feature!

Displays header text if `displayTitle` present. When `displayTitle` absent, hero image rendered.

#### PR Dependencies

- https://github.com/rocketsofawesome/thor/pull/82

#### Relevant Tickets

- [Finishes #167598393]
    - https://www.pivotaltracker.com/n/projects/2089973/stories/167598393

### CODE REVIEW

**Comment by virtuexru** [comment]
this looks good

File: src/modules/contentful/touts/contentfulTout.base.js [hunk 1]
```diff
@@ -13,22 +13,48 @@ class BaseContentfulTout extends Component {
...
-            {heroButtons && heroButtons.map((button) => {
-              return(
-                <ContentfulRenderer {...button} key={button.sys.id} />
-              )
-            })}
+
+    // If searchTerm or displayTitle present, do not render hero image
+    if (searchTerm) {
+      let searchTitle = `You searched ‘${searchTerm}’`
+      if (!productsFound) searchTitle = `Aw, shucks! 0 results for your search ‘${searchTerm}.’`
```
Comment (virtuexru) [inline src/modules/contentful/touts/contentfulTout.base.js]: not sure how i feel about these one liners as opposed to ` if () { ... }` multiline brackets we use everywhere else but ill leave that open to discussion

File: src/modules/contentful/touts/contentfulTout.base.js [hunk 2]
```diff
@@ -13,22 +13,48 @@ class BaseContentfulTout extends Component {
...
           </div>
         </div>
-      </ContentfulRenderer>
-    )
+      )
+    } else if (displayTitle) {
+      return (
+        <div className='default-shop-header'>
+          <div className='default-shop-header-title-wrapper'>
+            <h1 className='default-shop-header-title'>{displayTitle}</h1>
```
Comment (virtuexru) [inline src/modules/contentful/touts/contentfulTout.base.js]: we could avoid this big `else if` block if we figure out the title (whether search, or display) before this and then set that variable to whatever it should be so we only render this block without an extra `if else` 

if that doesnt make sense let me know i can probably explain it better in person. basically `searchTitle` above and this `displayTitle` could be grabbed from one variable that would be set beforehand (before the return ()`

File: src/modules/contentful/touts/contentfulTout.base.js (hunk 1 above)
Comment (alexrockets) [reply src/modules/contentful/touts/contentfulTout.base.js]: Good to keep it consistent with rest of codebase, we can even customize a linter to catch situations like this

File: src/modules/contentful/touts/contentfulTout.base.js (hunk 2 above)
Comment (alexrockets) [reply src/modules/contentful/touts/contentfulTout.base.js]: TY, makes sense!

**Pull request merged**